Add initializer that takes a NavigationStack as parameter (#37)
matteopuc/swiftui-navigation-stack: README.md

attaching the easing directly to the transition. **Don't do this**. SwiftUI has still some problems with implicit animations attached to transitions, so it may not work. For example, implicit animations attached to a .slide transition won't work.

> ## NavigationStack injection

Reviewer: Let's move this entire section between the `Pop` section and the `Important` section because I've just realised we only mention the `NavigationStack` (and the programmatic navigation) in a section below. Here it's still confusing.

> If you want to use the NavigationStack also outside the views hierachy you need to create you custom NavigationStack and pass it as parameter to the NavigationStackView.

Reviewer: I'd change this to:

```suggestion
By default you can programmatically push and pop only inside the `NavigationStackView` hierarchy (by accessing the `NavigationStack` environment object). If you want to use the `NavigationStack` outside the `NavigationStackView` you need to create your own `NavigationStack` (wherever you want) **and pass it as parameter to the `NavigationStackView`**.
```

This is useful when you want to _decouple your routing logic from views by using your own router class_, for example:

```
class MyRouter {
    let navStack: NavigationStack

    init(navStack: NavigationStack) {
        self.navStack = navStack
    }

>     func rootView() {

Reviewer: There's a compile time error here. Missing return type? 🤔

        if userIsLoggedIn() {
            return HomeScreen()
        } else {
            return LoginScreen()
        }
    }

    func toLogin() {
        self.navStack.push(LoginScreen())
    }

>     //...

Reviewer: ```suggestion
    func toSignUp() {
         self.navStack.push(SignUpScreen())
     }
```

}

> struct ContentView: View {

Reviewer: To be compliant with the other examples in the documentation.

```suggestion
struct RootView: View {
```

>     let navStack: NavigationStack

Reviewer: Not needed.
```suggestion
```

    let router: MyRouter

    var body: some View {

>         NavigationStackView(navigationStack: navStack) {

Reviewer: ```suggestion
        NavigationStackView(navigationStack: router.navStack) {
```

            router.rootView()
        }
    }
}
```

> Setup the ContentView in your SceneDelegate (or similarly in the App class)
> 
> ```
> // Create the SwiftUI view that provides the window contents.
> let contentView = ContentView(navStack: Injector.shared.navStack,
>                                 router: Injector.shared.router)
> 
> // Use a UIHostingController as window root view controller.
> if let windowScene = scene as? UIWindowScene {
>     let window = UIWindow(windowScene: windowScene)
>     window.rootViewController = UIHostingController(rootView: contentView)
>     self.window = window
>     window.makeKeyAndVisible()
> }
> ```

Reviewer: I'd remove this since users can inject the `NavigationStack` wherever they want. This example is based on some specific patterns.
```suggestion
```
